kinetics_MO.py

```diff
@@ -0,0 +1,316 @@
+        init_cov_y = pickle.load(f)
+
+    # Round possible float solution to be integer 
+    for i in range(num_total):
+        for j in range(num_total):
+            if init_cov_y[i][j] > 0.99:
+                init_cov_y[i][j] = int(1)
+            else:
+                init_cov_y[i][j] = int(0)
+                
+    # initialize total manual number 
+    total_manual_init = 0 
+    # initialize the DCM installation flags
+    dynamic_install_init = [0,0,0]
+
+    # round solutions
+    # if floating solutions, if value > 0.01, we count it as an integer decision that is 1 or positive
+    for i in range(num_static,num_total):
+        if init_cov_y[i][i] > 0.01:
+            total_manual_init += 1 
+            
+            # identify which DCM this timepoint belongs to, turn the installation flag to be positive 
+            i_pos = int((i-num_static)/Nt)
+            dynamic_install_init[i_pos] = 1
+            
+    # compute total measurements number, this is for integer cut
+    total_measure_init = sum(init_cov_y[i][i] for i in range(num_total))
+            
+    # initialize cost, this cost is calculated by the given initial solution
+    cost_init = sum(dynamic_install_init)*200+total_manual_init*400 + (init_cov_y[0][0]+init_cov_y[1][1]+init_cov_y[2][2])*2000
+
+    # read FIM, initialize FIM and logdet
+    with open(fim_init_file, 'rb') as f:
+        fim_prior = pickle.load(f)
+        
+    # initialize FIM with a small element 
+    for i in range(4):
+        fim_prior[i][i] += small_element
+
+    # this time is to evaluate model creating time
+    t1 = time.time()
+
+    if not update_model:
+        # create model
+        mod = calculator.continuous_optimization(mixed_integer=mip_option, 
+                            obj=objective, 
+                            mix_obj = mix_obj_option, 
+                            alpha = alpha_opt,
+                            fixed_nlp = fixed_nlp_opt,
+                            fix=fix_opt, 
+                            upper_diagonal_only=sparse_opt, 
+                            num_dynamic_t_name = num_dynamic_time, 
+                            budget=budget_opt,
+                            init_cov_y= init_cov_y,
+                            initial_fim = fim_prior,
+                            dynamic_install_initial = dynamic_install_init, 
+                            total_measure_initial = total_measure_init, 
+                            static_dynamic_pair=static_dynamic,
+                            time_interval_all_dynamic = time_interval_for_all,
+                            total_manual_num_init=total_manual_init,
+                            cost_initial = cost_init, 
+                            FIM_diagonal_small_element=small_element,
+                            print_level=1)
+        
+    else:
+        update_model.budget = budget_opt
+
+    t2 = time.time()
+    mod = calculator.solve(mod, mip_option=mip_option, objective = objective)
```

Comment:
I added check sentences in the end of the solver function to check the termination condition and solver status. If not converged to optimal, it will print sentences to tell the status when the `print_level` is >= 1. Do we need to store the exit status to the object itself or store to like a list?